git-make-release

```diff
@@ -41,40 +14,44 @@ if [ -n "`git status --porcelain -uno`" ]; then
 fi
 
 # obtain the latest tag and version
-IS_TAGGED=1
-LAST_TAG=`git describe --abbrev=0 --tags` || IS_TAGGED=0
-if [ $IS_TAGGED == 1 ]; then
-  echo "Last tag: ${LAST_TAG}"
-  LAST_VERSION=`echo $LAST_TAG | sed -e 's|^v||'`
-  echo "Last version: ${LAST_VERSION}"
-
-  SHOW_CHANGES_COMMAND="${script_home}/git-show-changes-since-last-tag"
-else
-  LAST_VERSION="initial commit"
-  SHOW_CHANGES_COMMAND="git log --graph"
-  echo "No previous tagged version found."
-fi
-
-# obtain the change log since the latest tag
-echo "Changes since ${LAST_VERSION}:"
-CHANGES=`mktemp`
-echo "============================================================"
-${SHOW_CHANGES_COMMAND} | grep -v "^commit " | grep -v "Author: " | grep -v "Date: " | sed -e 's|^ *||' | tee $CHANGES
-echo "============================================================"
+LAST_TAG=`git tag | tail -n1`
+echo "Last tag: ${LAST_TAG}"
+LAST_VERSION=`echo $LAST_TAG | sed -e 's|^v||'`
+echo "Last version: ${LAST_VERSION}"
 
 # extract new version from CMakeLists.txt
-MAJOR=`grep -i 'set(${PROJECT_NAME}_MAJOR_VERSION' ${REPO_TOPLEVEL}/CMakeLists.txt | sed -e 's|^set(${PROJECT_NAME}_MAJOR_VERSION ||i' -e 's|)$||'`
-MINOR=`grep -i 'set(${PROJECT_NAME}_MINOR_VERSION' ${REPO_TOPLEVEL}/CMakeLists.txt | sed -e 's|^set(${PROJECT_NAME}_MINOR_VERSION ||i' -e 's|)$||'`
-PATCH=`grep -i 'set(${PROJECT_NAME}_PATCH_VERSION' ${REPO_TOPLEVEL}/CMakeLists.txt | sed -e 's|^set(${PROJECT_NAME}_PATCH_VERSION ||i' -e 's|)$||'`
+MAJOR=`grep -i 'set(${PROJECT_NAME}_MAJOR_VERSION' CMakeLists.txt | sed -e 's|^set(${PROJECT_NAME}_MAJOR_VERSION ||i' -e 's|)$||'`
+MINOR=`grep -i 'set(${PROJECT_NAME}_MINOR_VERSION' CMakeLists.txt | sed -e 's|^set(${PROJECT_NAME}_MINOR_VERSION ||i' -e 's|)$||'`
+PATCH=`grep -i 'set(${PROJECT_NAME}_PATCH_VERSION' CMakeLists.txt | sed -e 's|^set(${PROJECT_NAME}_PATCH_VERSION ||i' -e 's|)$||'`
+
+if [ -z "$MAJOR" -o -z "$MINOR" -o -z "$PATCH" ]; then
+  echo "Cannot detect version from CMakeLists.txt!"
+  echo -n "Enter new MAJOR version: "
+  read MAJOR
+  echo -n "Enter new MINOR version: "
+  read MINOR
+  echo -n "Enter new PATCH version: "
+  read PATCH
+  if (( MAJOR < 10 )); then
+    MAJOR=0$MAJOR
+  fi
+  if (( MINOR < 10 )); then
+    MINOR=0$MINOR
+  fi
+  if (( PATCH < 10 )); then
+    PATCH=0$PATCH
+  fi
+fi
 
-# check that the new version is different from the current
 NEXT_VERSION=${MAJOR}.${MINOR}.${PATCH}
+echo "The new version will be: ${NEXT_VERSION}"
+
+# check that the new version is different from the current
 if [ "$NEXT_VERSION" == "$LAST_VERSION" ]; then
```

Comment:
This test falls short in case of backport branches because it compares to the latest tag, which probably is master and will always be different, even if the version in the branch has not been increased. Was not working before, so no regression here.

Reply:
I think it will fail later anyway, because you cannot create a tag which already exists. This check is merely for convenience in case one forgets to edit the CMakeLists.txt and update the version, which might be annoying to notice only after editing the release notes.

Reply:
Agreed, nothing much to gain nor to lose. Backports are rare, and mistakes in there even more.